### PR TITLE
feat(highlight): add an option to switch stripIndent

### LINKED ISSUE #5422: Hexo strips leading spaces in code blocks
### Check List

- [X] I have already read [Docs page](https://hexo.io/docs/) & [Troubleshooting page](https://hexo.io/docs/troubleshooting).
- [X] I have already searched existing issues and they are not help to me.
- [X] I examined error or warning messages and it's difficult to solve.
- [X] I am using the [latest](https://github.com/hexojs/hexo/releases) version of Hexo. (run `hexo version` to check)
- [X] My Node.js is matched [the required version](https://hexo.io/docs/#Required-Node-js-version).

### Expected behavior

Leading spaces in code blocks are not stripped.

### Actual behavior

Leading spaces in code blocks are stripped.

### How to reproduce?

Render the following lines:
``````````markdown
This is a paragraph.
```
    This is a code block
    with 4 spaces
    at the beginning of
    each line.
```
``````````

Here is [a live example on my blog](https://triplecamera.github.io/renderer-test/).

### Is the problem still there under `Safe mode`?

Cannot generate pages at all under safe mode.

### Your Node.js & npm version

```text
v20.11.0
10.2.4
```

### Your Hexo and Plugin version

```text
hexo-site@0.0.0 D:\GitHub\triplecamera.github.io
├── hexo-generator-archive@2.0.0
├── hexo-generator-category@2.0.0
├── hexo-generator-index@3.0.0
├── hexo-generator-tag@2.0.0
├── hexo-renderer-ejs@2.0.0
├── hexo-renderer-pandoc@0.4.0
├── hexo-renderer-stylus@3.0.1
├── hexo-server@3.0.0
├── hexo-theme-next@8.19.1
└── hexo@7.1.1
```

### Your `package.json`

See <https://github.com/TripleCamera/triplecamera.github.io>.

### Your site's `_config.yml` (Optional)

See <https://github.com/TripleCamera/triplecamera.github.io>.

### Others

_No response_

### PR DESCRIPTION
**Pull request by uiolee**
need hexojs/hexo-util#398
fix #5422

## What does it do?

add an option to disable stripIndent
issue #5422

## Pull request tasks

- [ ] Add test cases for the changes.
- [ ] Passed the CI test.

### CODE REVIEW

File: lib/hexo/default_config.ts [hunk 1]
```diff
@@ -49,14 +49,14 @@ export = {
     hljs: false,
     line_threshold: 0,
     first_line_number: 'always1',
-    disable_strip_indent: false
+    strip_indent: false
```
Comment (yoshinorin) [inline lib/hexo/default_config.ts:52]: Default should be `true` ?

Comment (uiolee) [reply lib/hexo/default_config.ts:52]: yes. I forget to change it

File: lib/hexo/default_config.ts [hunk 2]
```diff
@@ -48,13 +48,15 @@ export = {
     language_attr: false,
     hljs: false,
     line_threshold: 0,
-    first_line_number: 'always1'
+    first_line_number: 'always1',
+    strip_indent: false
```
Comment (uiolee) [inline lib/hexo/default_config.ts:52]:
```suggestion
    strip_indent: true
```

File: lib/hexo/default_config.ts [hunk 3]
```diff
@@ -48,13 +48,15 @@ export = {
...
   },
   prismjs: {
     preprocess: true,
     line_number: true,
     tab_replace: '',
-    exclude_languages: []
+    exclude_languages: [],
+    strip_indent: false
```
Comment (uiolee) [inline lib/hexo/default_config.ts:59]:
```suggestion
    strip_indent: true
```

**Comment by yoshinorin** [comment]
We need to release a new version of https://github.com/hexojs/hexo-util before merging this PR, right?

**Comment by uiolee** [comment]
yes, also need to update hexo-util in package.json

**Review by yoshinorin — APPROVED** [review]
LGTM 👍

**Pull request merged**